deltacat/utils/ray_utils/concurrency.py

```diff
@@ -49,20 +48,14 @@ def invoke_parallel(
     if max_parallelism is not None and max_parallelism <= 0:
         raise ValueError(f"Max parallelism ({max_parallelism}) must be > 0.")
     pending_ids = []
+    remaining_refs = []
+
     for i, item in enumerate(items):
-        if max_parallelism is not None and len(pending_ids) > max_parallelism:
-            # Some tasks return multiple values while others have only return one.
-            # For multiple return values we flatten the list of pending futures.
-            # TODO (pdames): Support tasks with an unbound number of return values.
-            if isinstance(pending_ids[0], list):
-                ray.wait(
-                    list(itertools.chain(*pending_ids)),
-                    num_returns=int(
-                        len(pending_ids[0]) * (len(pending_ids) - max_parallelism)
-                    ),
-                )
-            else:
-                ray.wait(pending_ids, num_returns=len(pending_ids) - max_parallelism)
+        if max_parallelism is not None and len(remaining_refs) > max_parallelism:
+            _, remaining_refs = ray.wait(
```

Comment:
Do we need both `pending_ids` and `remaining_refs`?

Reply:
Yeah, because pending_ids is what we return.

Reply:
To clarify, why not just re-use pending_ids for `ray.wait()`?

Reply:
the remaining_ids list become empty at the end. Hence, we cannot return it as a result.